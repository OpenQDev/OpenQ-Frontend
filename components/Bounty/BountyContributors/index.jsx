// Third party
import React, { useEffect, useState } from 'react';
import Link from 'next/link';

// Custom

const BountyContributors = ({ bounty }) => {
  const [contributors, setContributors] = useState({});

  useEffect(() => {
    let contribs = {};
    bounty?.deposits?.forEach((deposit) => {
      contribs[deposit.sender.id] = deposit.volume;
    });

    setContributors(contribs);
  }, []);

  return (
    <>
      <div className='font-bold text-xl'>Contributors</div>
      <div>Total: {Object.keys(contributors).length}</div>
      <br />
      {/*Each contributor account should be clickable so people can see other things they've funded, bounties created, etc. more social*/}
      {Object.keys(contributors).map((contributor) => {
        return (
          <Link href={`/user/${contributor}`} key={contributor}>
            {contributor}
          </Link>
        );
      })}
    </>
  );
};

export default BountyContributors;
