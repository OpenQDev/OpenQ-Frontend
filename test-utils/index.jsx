// Third party
import React from 'react';
// Cusotm
import { render, renderHook } from '@testing-library/react';
import { cleanup } from '@testing-library/react';

import StoreProvider from '../store/Store/TestStoreProvider';
import AuthProvider from '../store/AuthStore/TestAuthProvider';
import FundProvider from '../components/FundBounty/FundStore/TestFundProvider';
import TokenProvider from '../components/TokenSelection/TokenStore/TokenProvider';

// Add in any providers here if necessary:
// test-utils.js
// (ReduxProvider, ThemeProvider, etc)
afterEach(() => {
  cleanup();
});

const customRender = (ui, options = {}, storeProps, authProps) => {
  const IntersectionObserverMock = vi.fn(() => ({
    disconnect: vi.fn(),
    observe: vi.fn(),
    takeRecords: vi.fn(),
    unobserve: vi.fn(),
  }));
  vi.stubGlobal('IntersectionObserver', IntersectionObserverMock);
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <TokenProvider>{children}</TokenProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return render(ui, { wrapper: Providers, ...options });
};
const customHookRender = (hook, options = {}, storeProps, authProps, fundProps) => {
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <FundProvider StoreProps={fundProps}>
            <TokenProvider>{children}</TokenProvider>
          </FundProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return renderHook(hook, { wrapper: Providers, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, customHookRender as hookRender };
