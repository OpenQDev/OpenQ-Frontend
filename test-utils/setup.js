import { expect, beforeEach } from 'vitest';
import { cleanup } from '@testing-library/react';
import matchers from '@testing-library/jest-dom/matchers';

expect.extend(matchers);
beforeEach(() => {
  cleanup();
});
process.on('unhandledRejection', (reason) => {
  // eslint-disable-next-line no-console
  console.log(`FAILED TO HANDLE PROMISE REJECTION`);
  throw reason;
});
