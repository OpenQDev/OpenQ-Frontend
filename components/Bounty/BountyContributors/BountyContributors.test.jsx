import React from 'react';
import BountyContributors from '.';
import renderer from 'react-test-renderer';
import Constants from '../../../test-utils/constant';
import { render, screen } from '../../../test-utils';

describe('BountyContributors Snapshot', () => {
  const bounty = Constants.bounty;

  it('should match DOM Snapshot and 0 contribs', () => {
    const tree = renderer.create(<BountyContributors />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('should render the list of 1 contrib', () => {
    render(<BountyContributors bounty={bounty} />);
    expect(screen.findByText(/Total: 1/i));
  });
});
