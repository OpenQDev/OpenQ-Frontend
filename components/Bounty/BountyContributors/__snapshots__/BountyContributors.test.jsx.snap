// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BountyContributors Snapshot > should match DOM Snapshot and 0 contribs 1`] = `
[
  <div
    className="font-bold text-xl"
  >
    Contributors
  </div>,
  <div>
    Total: 
    0
  </div>,
  <br />,
]
`;
